solr/contrib/s3-repository/src/java/org/apache/solr/s3/S3StorageClient.java

```diff
@@ -73,48 +80,51 @@
   // Error messages returned by S3 for a key not found.
   private static final Set<String> NOT_FOUND_CODES = Set.of("NoSuchKey", "404 Not Found");
 
-  private final AmazonS3 s3Client;
+  private final S3Client s3Client;
 
   /** The S3 bucket where we read/write all data. */
   private final String bucketName;
 
   S3StorageClient(
-      String bucketName, String region, String proxyHost, int proxyPort, String endpoint) {
-    this(createInternalClient(region, proxyHost, proxyPort, endpoint), bucketName);
+      String bucketName,
+      String region,
+      String proxyUrl,
+      boolean proxyUseSystemSettings,
+      String endpoint) {
+    this(createInternalClient(region, proxyUrl, proxyUseSystemSettings, endpoint), bucketName);
   }
 
   @VisibleForTesting
-  S3StorageClient(AmazonS3 s3Client, String bucketName) {
+  S3StorageClient(S3Client s3Client, String bucketName) {
     this.s3Client = s3Client;
     this.bucketName = bucketName;
   }
 
-  private static AmazonS3 createInternalClient(
-      String region, String proxyHost, int proxyPort, String endpoint) {
-    ClientConfiguration clientConfig = new ClientConfiguration().withProtocol(Protocol.HTTPS);
-
+  private static S3Client createInternalClient(
+      String region, String proxyUrl, boolean proxyUseSystemSettings, String endpoint) {
+    ApacheHttpClient.Builder sdkHttpClientBuilder = ApacheHttpClient.builder();
```

Comment:
I think I had to switch to using the ApacheHttpClient for the S3Client, otherwise we wouldn't be able to use a proxy.

The new ApacheHttpClient also lets us use system proxy settings now, so I've enabled that by default. But users can turn it off if they desire in the solr.xml..